virttest/utils_vsock.py

```diff
@@ -0,0 +1,46 @@
+import os
+import logging
+import fcntl
+import struct
+import errno
+
+from avocado.core import exceptions
+from avocado.utils import linux_modules
+
+from virttest import arch
+
+VSOCK_PATH = '/dev/vhost-vsock'
+
+
+def load_vsock_module():
+    """ Load vhost_vsock module, to enable vsock function """
+    if not os.path.exists(VSOCK_PATH):
+        logging.info('vsock module was not loaded, loading it...')
+        if not linux_modules.load_module('vhost_vsock'):
+            raise exceptions.TestError(
+                'Failed on loading module vhost_vsock.')
+
+
+def get_guest_cid(guest_cid):
+    """
+    Get an unused guest cid from system
+
+    :param guest_cid: Requested guest cid
+    :return Available guest cid
+    """
+    vsock_fd = os.open(VSOCK_PATH, os.O_RDWR)
+    while guest_cid:
+        cid_c = struct.pack('L', guest_cid)
+        try:
+            fcntl.ioctl(
+                vsock_fd, arch.VHOST_VSOCK_SET_GUEST_CID, cid_c)
+        except IOError as e:
+            if e.errno == errno.EADDRINUSE:
+                guest_cid += 1
+                continue
+            else:
+                os.close(vsock_fd)
+                raise e
+        else:
+            os.close(vsock_fd)
```

Comment:
Prefer move it to finally.

Reply:
@yanan-fu I think no, there is one condition that should not be closed, at L#40.

Reply:
@vivianQizhu  yes,  right,  need continue to get cid.

Reply:
It can be done by doing like below:
```python
vsock_fd = os.open(VSOCK_PATH, os.O_RDWR)
try:
    while guest_cid:
        ...
finally:
    os.close(vsock_fd)
```